### PR TITLE
sesh: init at 0.12.0

### PR DESCRIPTION
**Pull request by Meliketoaste**
## Description of changes
Added the package https://github.com/joshmedeski/sesh. It a session manager and the sequel to https://github.com/joshmedeski/t-smart-tmux-session-manager. 

## Things done
- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [X] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/se/sesh/package.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "sesh";
+  version = "be2badd29206f66803b76b62b23f6f5e05befe08";
```
Comment (Ligthiago) [inline pkgs/by-name/se/sesh/package.nix:5]: Please use tagged version

```suggestion
  version = "0.12.0";
```

File: pkgs/by-name/se/sesh/package.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+  src = fetchFromGitHub {
+    owner = "joshmedeski";
+    repo = "sesh";
+    rev = "be2badd29206f66803b76b62b23f6f5e05befe08";
+    sha256 = "sha256-8gn4YdKHurPbPuJ+AeCB7x9IDxTtHGpYUZCKlSdujcs=";
```
Comment (Ligthiago) [inline pkgs/by-name/se/sesh/package.nix:11]:
```suggestion
    rev = "v${version}";
    sha256 = "sha256-m/EcWh4wfna9PB/NN+MCRUsz5Er0OZ70AAumlKdrm/s=";
```

File: pkgs/by-name/se/sesh/package.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
+  vendorHash = "sha256-zt1/gE4bVj+3yr9n0kT2FMYMEmiooy3k1lQ77rN6sTk=";
+
+  meta = with lib; {
+    description = "Smart session manager for the terminal.";
+    homepage = "https://github.com/joshmedeski/sesh";
+    license = licenses.mit;
+    maintainers = with maintainers; [ Meliketoaste ];
```
Comment (Ligthiago) [inline pkgs/by-name/se/sesh/package.nix:20]: Add mainProgram

```suggestion
    maintainers = with maintainers; [ Meliketoaste ];
    mainProgram = "sesh";
```

File: pkgs/by-name/se/sesh/package.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  };
+
```
Comment (Ligthiago) [inline pkgs/by-name/se/sesh/package.nix:22]:
```suggestion
```

**Review by Ligthiago — CHANGES_REQUESTED** [review]
Please, refer to https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

Rename PR to "sesh: init at 0.12.0". Squash commits together, one commit per logical unit (package in this case). Add yourself to the maintainers list

**Comment by Meliketoaste** [comment]
nvm i now noticed that someone else packaged it already.

**Pull request closed without merging**